packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -953,14 +954,21 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
       poppedItem = value.pop()
     }
 
-    this.trySetState({ value })
-
     if (getA11ySelectionMessage && getA11ySelectionMessage.onRemove) {
       this.setA11yStatus(getA11ySelectionMessage.onRemove(poppedItem))
     }
 
-    // we don't have event for it, but want to keep the event handling interface, event is empty.
-    _.invoke(this.props, 'onSelectedChange', {}, { ...this.props, value })
+    this.setStateAndInvokeHandler({ value }, 'onSelectedChange')
+  }
+
+  /**
+   * Calls trySetState (for autoControlledProps) and invokes event handler exposed to user.
+   * We don't have the event object for most events coming from Downshift se we send an empty event
+   * because we want to keep the event handling interface
+   */
+  private setStateAndInvokeHandler = (newState: Partial<DropdownState>, eventName: string) => {
```

Comment:
1. Actually, this is a `handlerName`.
2. `handlerName` should be defined in props.
3. It's more intuitive to have it on first place `setStateAndInvokeHandler('onOpenChange', { open: true })`.
4. It should be named as `trySetStateAndInvokeHandler()`.
5. Let's allow to pass event:  `setStateAndInvokeHandler('onOpenChange', null, { open: true })`.

```suggestion
  private trySetStateAndInvokeHandler = (handlerName: keyof DropdownProps, event: any, newState: Partial<DropdownState>) => {
```

Reply:
great suggestions, done, thanks!